backend/battle/tests/test_api.py

```diff
@@ -0,0 +1,354 @@
+        self.assertEqual(response.status_code, 200)
+
+    def test_endpoint_returns_empty_list(self):
+        battle = baker.make('battle.Battle', creator=self.user_1)
+        response = self.auth_client.get(reverse('battle-list'))
+        matching_battles = BattleSerializer([battle], many=True)
+        self.assertEqual(response.status_code, 200)
+        self.assertCountEqual(matching_battles.data, response.json())
+
+    def test_endpoint_returns_one_battle_in_list(self):
+        battle = baker.make('battle.Battle', creator=self.user_1)
+        response = self.auth_client.get(reverse('battle-list'))
+        matching_battles = BattleSerializer([battle], many=True)
+        self.assertEqual(response.status_code, 200)
+        self.assertCountEqual(matching_battles.data, response.json())
+
+    def test_endpoint_returns_few_battle_ids(self):
+        battle = baker.make('battle.Battle', creator=self.user_1, _quantity=3)
+        response = self.auth_client.get(reverse('battle-list'))
+        matching_battles = BattleSerializer(battle, many=True)
+        self.assertEqual(response.status_code, 200)
+        self.assertCountEqual(matching_battles.data, response.json())
+
+    def test_endpoint_returns_a_lot_battle_ids(self):
+        battle = baker.make('battle.Battle', creator=self.user_1, _quantity=100)
+        response = self.auth_client.get(reverse('battle-list'))
+        matching_battles = BattleSerializer(battle, many=True)
+        self.assertEqual(response.status_code, 200)
+        self.assertCountEqual(matching_battles.data, response.json())
+
+    def test_endpoint_returns_an_error_when_the_user_is_not_logged(self):
+        self.auth_client.logout()
+        response = self.client.get(reverse('battle-list'))
+        self.assertEqual(response.status_code, 403)
+
+    def test_endpoint_diff_to_verify_it_returns_exactly_len_of_battle_list(self):
+        battle = baker.make('battle.Battle', creator=self.user_1, _quantity=10)
+
+        response_initial = self.auth_client.get(reverse('battle-list'))
+
+        self.assertCountEqual(BattleSerializer(battle, many=True).data, response_initial.json())
+
+        baker.make('battle.Battle', creator=self.user_1, _quantity=50)
+
+        all_battles = Battle.objects.filter(creator=self.user_1)
+
+        response_updated = self.auth_client.get(reverse('battle-list'))
+
+        self.assertCountEqual(
+            BattleSerializer(all_battles, many=True).data,
+            response_updated.json())
+
+
+class DeatilBattleEndpointTest(APITestCaseUtils):
+
+    def test_login_user_can_acess_battle_detail(self):
+        battle = baker.make("battle.Battle", creator=self.user_1, opponent=self.opponent)
+        response = self.auth_client.get(reverse('battle-detail', kwargs={'pk': battle.id}))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            BattleSerializer([battle], many=True).data[0]['creator'],
+            response.json()['id'])
```

Comment:
This is a bit confusing, I'm not sure what you are testing here. Is that if the creator has the same id of the battle? If yes, this doesn't look like the appropriate thing to test

Reply:
It's true, thanks for the advice :)